executor/status_handler.go

```diff
@@ -16,87 +16,87 @@ func (executor *Executor) podsStatusHandler(entityName string, namespace string,
 	msg = "pods restarting exceeded timout (15 min)"
 	start := time.Now()
 
-	objectName := ""
-	result = proto.AutomationFailed
-	targetPods = 0
-	var err error = nil
-	flag := false
-
-	if strings.ToLower(kind) == "deployment" {
-		objectName, targetPods, err = executor.deploymentsHandler(entityName, namespace)
-		if err != nil {
-			flag = true
+	for time.Now().Sub(start) < automationsExecutionTimeout {
 
-		}
+		objectName := ""
+		result = proto.AutomationFailed
+		targetPods = 0
+		var err error = nil
+		flag := false
 
-	} else if strings.ToLower(kind) == "statefulset" {
-		objectName, targetPods, err = executor.statefulsetsHandler(entityName, namespace)
-		if err != nil {
-			flag = true
+		time.Sleep(podStatusSleep)
 
-		}
+		if strings.ToLower(kind) == "deployment" {
+			objectName, targetPods, err = executor.deploymentsHandler(entityName, namespace)
+			if err != nil {
+				flag = true
 
-	} else if strings.ToLower(kind) == "daemonset" {
-		objectName, targetPods, err = executor.daemonsetsHandler(entityName, namespace)
-		if err != nil {
-			flag = true
+			}
 
-		}
+		} else if strings.ToLower(kind) == "statefulset" {
+			objectName, targetPods, err = executor.statefulsetsHandler(entityName, namespace)
```

Comment:
why you moved the other handlers in side the time iterations loop, know you are doing extra iteration for all types of controllers